crates/sui-types/src/object.rs

```diff
@@ -34,6 +33,9 @@ pub const OBJECT_START_VERSION: SequenceNumber = SequenceNumber::from_u64(1);
 #[derive(Eq, PartialEq, Debug, Clone, Deserialize, Serialize, Hash)]
 pub struct MoveObject {
     pub type_: StructTag,
+    /// Determines if it is usable with the PublicTransferObject
+    /// Derived from the type_
+    has_public_transfer: bool,
```

Comment:
Can we also add this field to [SuiMoveObject](https://github.com/MystenLabs/sui/blob/764f4fd9d2dff37be0345080ada8d103e87de23b/crates/sui-core/src/gateway_types.rs#L323)? `MoveObject` will get translated into `SuiMoveObject` in the RPC layer to make sure RPC has a stable API. Having `has_public_transfer` exposed to the RPC client will enable the client(e.g. wallet) to tell if the object is transferrable

cc @patrickkuo